File: growing-shop-java/src/main/java/com/example/growingshop/domain/auth/api/RoleController.java
package com.example.growingshop.domain.auth.api;

import com.example.growingshop.domain.auth.dto.RoleRequest;
import com.example.growingshop.domain.auth.dto.RoleResponse;
import com.example.growingshop.domain.auth.service.RoleService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/roles")
public class RoleController {
    private final RoleService roleService;

    @GetMapping
    public ResponseEntity<List<RoleResponse.RoleRes>> findAll() {
        return new ResponseEntity<>(roleService.findAll().toResponse(), HttpStatus.OK);
    }

    @PostMapping
    public ResponseEntity<RoleResponse.RoleRes> createRole(@RequestBody RoleRequest.CreateRole req) {
        return new ResponseEntity<>(
                RoleResponse.RoleRes.from(roleService.create(req)),
                HttpStatus.CREATED
        );
    }

    @DeleteMapping
    public void deleteRole(Long id) {
        roleService.delete(id);
    }

    @PutMapping("/policies")
    public ResponseEntity changePolicies(@RequestBody RoleRequest.ChangeRolePolicies req) {
        roleService.changePolicies(req);

        return ResponseEntity.noContent().build();
    }
}

File: growing-shop-api/auth/src/main/kotlin/com/example/growingshopauth/auth/api/RoleController.kt
package com.example.growingshopauth.auth.api

import com.example.growingshopauth.auth.domain.toResponse
import com.example.growingshopauth.auth.dto.RoleRequest
import com.example.growingshopauth.auth.dto.RoleResponse
import com.example.growingshopauth.auth.service.RoleService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/roles")
class RoleController(
    private val roleService: RoleService
) {

    @GetMapping
    fun findAll(): ResponseEntity<List<RoleResponse.RoleRes>> = ResponseEntity(
        roleService.findAll().toResponse(),
        HttpStatus.OK
    )

    @PostMapping
    fun create(@RequestBody req: RoleRequest.CreateRole): ResponseEntity<RoleResponse.RoleRes> = ResponseEntity(
        RoleResponse.RoleRes.from(roleService.create(req)),
        HttpStatus.CREATED
    )

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long): ResponseEntity<Any> {
        roleService.delete(id)

        return ResponseEntity(HttpStatus.OK)
    }
}
